src/modm/platform/flash/stm32/flash.cpp.in

```diff
@@ -92,8 +92,20 @@ Flash::erase(uint8_t index)
 			((index << FLASH_CR_SNB_Pos) & FLASH_CR_SNB_Msk);
 %% else
 %% if family in ["g0","g4"]
+%% if dual_bank
+%% if family == "g0"
+	const bool index_is_on_second_bank{index >= 256};
+%% else
+	const bool index_is_on_second_bank{index >= (Size/2048/2)};
+%% endif
+	const uint8_t page = index_is_on_second_bank ? index - Size/2048/2 : index;
+	FLASH->CR = FLASH_CR_STRT | FLASH_CR_PER |
+			((index_is_on_second_bank << FLASH_CR_BKER_Pos) & FLASH_CR_BKER_Msk) |
+			((page << FLASH_CR_PNB_Pos) & FLASH_CR_PNB_Msk);
+%% else
```

Comment:
I'm not super happy with this, because it doesn't adhere to the [category 3 flash page numbering in RM0440, which goes from 0...127 on both banks](https://www.st.com/resource/en/reference_manual/rm0440-stm32g4-series-advanced-armbased-32bit-mcus-stmicroelectronics.pdf#page=95).
I think the `Flash::erase` function should actually have a second parameter where the user can specify the bank.
However, that would break the `Flash::getPage` function and potentially other things I don't want to fix right now.

Also, the flash page numbering on the g0 devices is even more crude (bank 1 goes from 0...128 or so) and bank 2 goes from 256... on. :brain: 

Does anyone have a better idea how to do this?

Reply:
The flash page size differs depending on the dual bank option byte? Ie. if dual bank, it's 2048B, otherwise it's 4096B??? Then we need to modify `shift` to be `(FLASH->OPTR & FLASH_OPTR_DBANK) ? 11 : 12)` in `getPage()`, `getOffset()` and `getSize()`, which is ugly, but would solve the problem?

Reply:
Yes :-D looks like thats the case!

Reply:
I think the CI fails from your changes are due to the `FLASH_CR_BKER` and `FLASH_OPTR_DBANK` only beeing available on G4 devices with category 3 flash. On G4 devices with category 2 and 4 flash, there is no dual bank option :brain:

Reply:
Ok, I implemented it that way. Apparently only G47x and G48x are category 3 devices, so let's see if I fixed it.

Reply:
And G0Bx and G0Cx are the only dual-bank G0 devices.